### PR TITLE
Avoid squeezing original weight tensors with leading dim 1

### PR DESCRIPTION
**Pull request by cjluo-nv**
## What does this PR do?

bug fix

**Overview:** ?

Phi4MM has weights with leading dim 1. Squeezing it will yield to shape mismatch with the original weight.

## Testing

scripts/huggingface_example.sh --model /models/microsoft/Phi-4-multimodal-instruct/ --quant fp8 --export_fmt hf --trust_remote_code

and check the weights shape

## Summary by CodeRabbit

* **Bug Fixes**
  * Avoids unintended collapsing of leading singleton dimensions for certain 3D tensors during export/quantization; only scale-related tensors are collapsed.
  * Reduces shape mismatches and runtime errors when loading, converting, or serving quantized models, improving checkpoint stability and compatibility.

### CODE REVIEW

File: modelopt/torch/export/quant_utils.py [hunk 1]
```diff
@@ -869,11 +869,6 @@ def postprocess_state_dict(state_dict: dict, maxbound: float, quantization: str
                 post_state_dict[prefix + new_suffix] = value
                 break
 
-    # Squeeze tensors with a leading dimension of 1
```
Comment (Edwardf0t1) [inline modelopt/torch/export/quant_utils.py:872]: Can you specify what errors you get for phi4-mm?

We needed this for sticking with our ckpt format and I remember if we don’t squeeze it will break some deployment flow with TRT-LLM.
It works for all our supported models so far.

**Comment by ChenhanYu** [comment]
https://gitlab-master.nvidia.com/omniml/modelopt/-/pipelines/34670934
